### PR TITLE
Create a library that works with a browser

### PR DESCRIPTION
**Pull request by MichaelKim20**
I replaced the sodium library with a version that uses JavaScript and WASM.
I replaced it with JSBI.BigInt for browsers that do not support BigInt.
~I replaced writeBigUInt64LE with writeUInt32LE~
Replace with using bit operation instead of writeBigUInt64LE because some browsers do not support writeBigUInt64LE
I create a library for the browsers using the webpack

I did a test on four kinds of browsers. I have confirmed that all of them are operating normally.

**Chrome**

**Safari**

**FireFox**

**Microsoft Edge**

Test code is https://github.com/MichaelKim20/boa-sdk-ts-test
``` JavaScript
            writeMessage("[Test for Hash]");
            var h = BoaSdk.hash(stringToUint8Array('abc'));
            if (h.toString() === '0x239900d4ed8623b95a92f1dba88ad31895cc3345ded552c22d79ab2a39c5877' +
                'dd1a2ffdb6fbb124bb7c45a68142f214ce9f6129fb697276a0d4d1c983fa580ba') {
                writeMessage("    => Hash Success");
            }
            else {
                writeMessage("    => Hash Fail");
            }

            writeMessage("[Test for makeUTXOKey]");
            var tx_hash = new BoaSdk.Hash();
            tx_hash.fromString('0x5d7f6a7a30f7ff591c8649f61eb8a35d034824ed5cd252c2c6f10cdbd223671' +
                '3dc369ef2a44b62ba113814a9d819a276ff61582874c9aee9c98efa2aa1f10d73');
            var hash = BoaSdk.makeUTXOKey(tx_hash, BoaSdk.JSBInt.BigInt(1));

            if (hash.toString() === '0x7c95c29b184e47fbd32e58e5abd42c6e22e8bd5a7e934ab049d21df545e09c2' +
                'e33bb2b89df2e59ee01eb2519b1508284b577f66a76d42546b65a6813e592bb84') {
                writeMessage("    => makeUTXOKey Success");
            }
            else {
                writeMessage("    => makeUTXOKey Fail");
            }

            writeMessage("[Test for verify]");
            var address = 'GDD5RFGBIUAFCOXQA246BOUPHCK7ZL2NSHDU7DVAPNPTJJKVPJMNLQFW';
            var signature = new BoaSdk.Signature();
            signature.fromString('0x01cd1b598618347c1d5df7ce25d94de41bac109699dc0c93bb01703b81' +
                'c7731503f7dc5f7f9a9375efb672c83669f5c74ba41d30e5a969d1ed06904eb3b0b6a7');
            let public_key = BoaSdk.PublicKey.fromString(address);
            if (public_key.verify(signature, stringToUint8Array('Hello World')))
            {
                writeMessage("    => verify Success");
            }
            else {
                writeMessage("    => verify Fail");
            }

        function stringToUint8Array(str) {
            var arr = [];
            for (var i = 0; i < str.length; i++)
                arr.push(str.charCodeAt(i));
            return new Uint8Array(arr);
        }
        function writeMessage(msg) {
            $('#target').append('<div>' + msg + '</div>');
        }
```
Relates to Issue #14

### CODE REVIEW

File: src/modules/data/Hash.ts [hunk 1]
```diff
@@ -99,7 +99,10 @@ export function hashMulti (source1: Buffer, source2: Buffer): Hash
 export function makeUTXOKey (h: Hash, index: bigint): Hash
 {
     let idx = Buffer.alloc(8);
-    idx.writeBigUInt64LE(index);
+    const big = Number(~~(index / BigInt(0x0100000000)));
```
Comment (AndrejMitrovic) [inline src/modules/data/Hash.ts]: What does `~~` do?

File: src/modules/data/Hash.ts [hunk 2]
```diff
@@ -99,7 +99,10 @@ export function hashMulti (source1: Buffer, source2: Buffer): Hash
 export function makeUTXOKey (h: Hash, index: bigint): Hash
```
Comment (AndrejMitrovic) [inline src/modules/data/Hash.ts]: > Replace writeBigUInt64LE with writeUInt32LE

There is no commit message. It should explain **why** the change was made.

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+module.exports = {
+    entry: {
+        "boa-sdk": "./src/index.ts",
+        "boa-sdk.min": "./src/index.ts",
+    },
+    devtool: 'source-map',
+    module: {
+        rules: [
+            {
+                test: /\.tsx?$/,
```
Comment (TrustHenry) [inline webpack.config.js:12]:
```suggestion
                test: /\.test.ts?$/,
```
The loader specified in the loader compiles the files that meet the test regular expression conditions.

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                use: 'ts-loader',
+                exclude: /node_modules/,
+            },
+        ],
+    },
+    resolve: {
+        extensions: [ '.tsx', '.ts', '.js' ],
```
Comment (TrustHenry) [inline webpack.config.js:19]: Does it include .tsx file?

File: src/modules/data/Hash.ts (hunk 1 above)
Comment (MichaelKim20) [reply src/modules/data/Hash.ts]: > What does `~~` do?

Two bitwiseNot.

Comment (AndrejMitrovic) [reply src/modules/data/Hash.ts]: Yes but I don't understand what the effect it has.

File: webpack.config.js (hunk 3 above)
Comment (MichaelKim20) [reply webpack.config.js:12]: `test: /\.ts?$/` 
I think this is right.

File: webpack.config.js (hunk 4 above)
Comment (MichaelKim20) [reply webpack.config.js]: Good catch!

File: src/modules/data/Hash.ts (hunk 1 above)
Comment (MichaelKim20) [reply src/modules/data/Hash.ts]: That's a good point.
I found better source code for this.
So I change it to something easier to understand and record the location of the original.
https://github.com/nodejs/node/blob/88fb5a5c7933022de750745e51e5dc0996a1e2c4/lib/internal/buffer.js#L573-L592

**Comment by MichaelKim20** [comment]
I checked that it works well on the safari of the iPhone.

File: src/modules/data/Hash.ts [hunk 5]
```diff
@@ -96,22 +97,37 @@ export function hashMulti (source1: Buffer, source2: Buffer): Hash
  * @returns Instance of Hash
  * See_Also https://github.com/bpfkorea/agora/blob/v0.x.x/source/agora/consensus/data/UTXOSetValue.d#L50-L53
  */
-export function makeUTXOKey (h: Hash, index: bigint): Hash
+export function makeUTXOKey (h: Hash, index: number|string|boolean|object): Hash
```
Comment (TrustHenry) [inline src/modules/data/Hash.ts:100]: `index` only supports `ulong` or `unit` type
https://github.com/bpfkorea/agora/blob/2fe227e8b670e3b407750fd1499370bee720980c/source/agora/consensus/data/UTXOSetValue.d#L39-L52

This makes sure the index is always of the same type,

Comment (MichaelKim20) [reply src/modules/data/Hash.ts:100]: This is where Stoa has already been proven.
And please refer to the test code.

Comment (MichaelKim20) [reply src/modules/data/Hash.ts:100]: I will change it as below.
export function makeUTXOKey (h: Hash, index: number|string): Hash
`string` can be used when entering 64Bit unsigned integer of 'index'.

**Pull request merged**